services/device-registry/src/test/java/org/eclipse/hono/deviceregistry/DeviceRegistryRestServerTest.java

```diff
@@ -0,0 +1,430 @@
+    private static FileBasedRegistrationService registrationService;
+    private static DeviceRegistryRestServer deviceRegistryRestServer;
+
+    @BeforeClass
+    public static void setUp(final TestContext context) throws Exception {
+        vertx = Vertx.vertx();
+
+        Future<String> setupTracker = Future.future();
+        setupTracker.setHandler(context.asyncAssertSuccess());
+
+        ServiceConfigProperties restServerProps = new ServiceConfigProperties();
+        restServerProps.setInsecurePortEnabled(true);
+        restServerProps.setInsecurePort(0);
+
+        RegistrationHttpEndpoint registrationHttpEndpoint = new RegistrationHttpEndpoint(vertx);
+        deviceRegistryRestServer = new DeviceRegistryRestServer();
+        deviceRegistryRestServer.addEndpoint(registrationHttpEndpoint);
+        deviceRegistryRestServer.setConfig(restServerProps);
+
+        FileBasedRegistrationConfigProperties regServiceProps = new FileBasedRegistrationConfigProperties();
+        registrationService = new FileBasedRegistrationService();
+        registrationService.setConfig(regServiceProps);
+        SignatureSupportingConfigProperties signatureSupportingConfProps = new SignatureSupportingConfigProperties();
+        signatureSupportingConfProps.setSharedSecret("DeviceRegistrySharedSecret_HasToBe32CharsOrLonger");
+        registrationService.setRegistrationAssertionFactory(
+                RegistrationAssertionHelperImpl.forSigning(vertx, signatureSupportingConfProps));
+
+        Future<String> restServerDeploymentTracker = Future.future();
+        vertx.deployVerticle(deviceRegistryRestServer, restServerDeploymentTracker.completer());
+        restServerDeploymentTracker.compose(s -> {
+            Future<String> registrationServiceDeploymentTracker = Future.future();
+            vertx.deployVerticle(registrationService, registrationServiceDeploymentTracker.completer());
+            return registrationServiceDeploymentTracker;
+        }).compose(c -> setupTracker.complete(), setupTracker);
+    }
+
+    @AfterClass
+    public static void tearDown(final TestContext context) throws Exception {
+        vertx.close(context.asyncAssertSuccess());
+    }
+
+    @After
+    public void clearRegistry() throws InterruptedException {
+        registrationService.clear();
+    }
+
+    private int getPort() {
+        return deviceRegistryRestServer.getInsecurePort();
+    }
+
+    @Test
+    public void testAddDevice(final TestContext context) throws Exception {
+        final String requestUri = String.format("/%s/%s", REGISTRATION_ENDPOINT, TENANT);
+        final String requestBody = "{\"" + FIELD_DEVICE_ID + "\":\"" + DEVICE_ID + "\",\"test\":\"test\"}";
+        final Async async = context.async();
+        vertx.createHttpClient().post(getPort(), HOST, requestUri).putHeader("content-type", CONTENT_TYPE_JSON)
+                .handler(response -> {
+                    context.assertEquals(201, response.statusCode());
```

Comment:
Maybe better use `HttpURLConnection.HTTP_CREATED`?
Same for all occurences of direct status code below....